src/processing/evm.rs

```diff
@@ -0,0 +1,198 @@
+use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use alloy::hex::FromHex;
+use alloy::primitives::FixedBytes;
+use arc_swap::Guard;
+use bigdecimal::ToPrimitive;
+use num_bigint::BigInt;
+use starknet::core::types::{BlockId, BlockTag, Felt, FunctionCall};
+use starknet::macros::selector;
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::{JsonRpcClient, Provider};
+
+use crate::config::{Config, EvmConfig};
+use crate::constants::EVM_TIME_SINCE_LAST_FEED_UPDATE;
+use crate::{config::get_config, error::MonitoringError};
+
+#[derive(Clone)]
+enum FeedType {
+    Unique(UniqueVariant),
+    Twap(TwapVariant),
+}
+
+impl FeedType {
+    async fn update_data(&self, chain: &EvmConfig, feed_id: FeedId) -> Result<(), MonitoringError> {
+        match self {
+            FeedType::Unique(unique_variant) => unique_variant.update_data(chain, feed_id).await,
+            FeedType::Twap(twap_variant) => twap_variant.update_data(chain, feed_id).await,
+        }
+    }
+}
+
+#[derive(Clone)]
+enum UniqueVariant {
+    SpotMedian,
+    PerpMedian,
+}
+
+#[derive(Clone)]
+enum TwapVariant {
+    SpotTwap = 0,
```

Comment:
let's be explicit on all variants for the number then
```rust
enum UniqueVariant {
    SpotMedian = 0,
    PerpMedian = 1,
}
```

Reply:
done